src/types/rule.ts

```diff
@@ -1,4 +1,6 @@
 import { DiagnosticSeverity } from '@stoplight/types';
+
+import { CasingRule } from '../functions/casing';
```

Comment:
This is an improvement but do we still need to do this? 

The whole export type Rule thing seems outdated now no?

Reply:
@philsturgeon As that looks like a different concern, I've created #898 to specifically address this.